### PR TITLE
Add ChromaDB support

### PR DESCRIPTION
**Pull request by Dev-Khant**
## Description

Add Chromadb to be used as vectordb for Mem0

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] Documentation update

## How Has This Been Tested?

- [x] Unit Test

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

## Maintainer Checklist

- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] Made sure Checks passed

### CODE REVIEW

**Review by kmitul — CHANGES_REQUESTED** [review]
Please add the `chromadb` dependency to the `pyproject.toml`. Here’s the suggested change to the `pyproject.toml` file:

```toml
[tool.poetry.dependencies]
python = "^3.8"
qdrant-client = "^1.9.1"
pydantic = "^2.7.3"
openai = "^1.33.0"
posthog = "^3.5.0"
chromadb = "^0.5.1" 
```

File: docs/components/vectordb.mdx [hunk 1]
```diff
@@ -0,0 +1,64 @@
+---
+title: 🗄 VectorDB
```
Comment (deshraj) [inline docs/components/vectordb.mdx:2]:
```suggestion
title: 🗄 Vector Databases
```

File: docs/components/vectordb.mdx [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+---
+
+## Overview
+
+Mem0 includes built-in support for various popular databases. Memory can utilize the database provided by the user, ensuring efficient use for specific needs.
+
+<CardGroup>
+    <Card title="Qdrant" href="#qdrant"></Card>
+    <Card title="Chromadb" href="#chromadb"></Card>
```
Comment (deshraj) [inline docs/components/vectordb.mdx:11]:
```suggestion
    <Card title="Chroma" href="#chroma"></Card>
```

File: docs/components/vectordb.mdx [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+            "port": 6333,
+        }
+    }
+}
+
+m = Memory.from_config(config)
+m.add("Likes to play cricket on weekends", user_id="alice", metadata={"category": "hobbies"})
+```
+
+## Chromadb
```
Comment (deshraj) [inline docs/components/vectordb.mdx:41]:
```suggestion
## Chroma
```

File: docs/components/vectordb.mdx [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+[Chromadb](https://www.trychroma.com/) is a high-performance vector database that is designed to work with large-scale datasets. It provides a scalable and efficient solution for storing and querying vector data.
```
Comment (deshraj) [inline docs/components/vectordb.mdx:43]:
```suggestion
[Chroma](https://www.trychroma.com/) is a high-performance vector database that is designed to work with large-scale datasets. It provides a scalable and efficient solution for storing and querying vector data.
```

File: mem0/utils/factory.py [hunk 5]
```diff
@@ -44,4 +44,18 @@ def create(cls, provider_name):
...
+    }
+
+    @classmethod
+    def create(cls, provider_name, config):
+        class_type = cls.provider_to_class.get(provider_name)
+        if class_type:
+            vector_store_instance = load_class(class_type)
+            return vector_store_instance(**config)
+        else:
+            raise ValueError(f"Unsupported VectorStore provider: {provider_name}")
```
Comment (deshraj) [inline mem0/utils/factory.py:61]:
```suggestion
            raise ValueError(f"Unsupported VectorStore provider: {provider_name}")

```

File: mem0/vector_stores/chroma.py [hunk 6]
```diff
@@ -0,0 +1,222 @@
...
+            filters (dict, optional): Filters to apply to the list. Defaults to None.
+            limit (int, optional): Number of vectors to return. Defaults to 100.
+
+        Returns:
+            list: List of vectors.
+        """
+        array = [[0 for _ in range(1536)] for _ in range(1536)]
+        results = self.collection.query(query_embeddings=array, where=filters, n_results=limit)
+        return [self._parse_output(results)]
+
```
Comment (deshraj) [inline mem0/vector_stores/chroma.py:222]:
```suggestion

```

File: pyproject.toml [hunk 7]
```diff
@@ -20,6 +20,7 @@ qdrant-client = "^1.9.1"
 pydantic = "^2.7.3"
 openai = "^1.33.0"
 posthog = "^3.5.0"
+chromadb = "^0.5.3"
```
Comment (deshraj) [inline pyproject.toml:23]: Let's not add it to our optional dependency list. Instead, use the try/except block around chroma import and show appropriate error to the user to install it properly.

Comment (deshraj) [reply pyproject.toml:23]: Moreover, Chroma is quite heavy and we don't want to bloat our package with that.

Comment (Dev-Khant) [reply pyproject.toml:23]: Alright will remove it

**Pull request merged**